File: src/main/java/org/fao/unredd/portal/admin/Context.java
package org.fao.unredd.portal.admin;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by sgiaccio on 11/02/15.
 *
 * Represents a portal Context definition
 */
public class Context implements Recursive {
    public String id;
    public String label;
    public Boolean active;
    public List<Layer> layers;
    public String infoFile;
    public String inlineLegendUrl;

    @Override
    public String toString() {
        return "Context{" +
                "id='" + id + '\'' +
                ", label='" + label + '\'' +
                ", active=" + active +
                ", layers=" + layers +
                ", infoFile='" + infoFile + '\'' +
                ", inlineLegendURL='" + inlineLegendUrl + '\'' +
                '}';
    }
}

File: src/main/javascript/app/models/context.js
import DS from 'ember-data';

export default DS.Model.extend({
  label: DS.attr('string'),
  active: DS.attr('boolean'),
  layers: DS.hasMany('layer', { async: true }),
  infoFile: DS.attr('string'),
  inlineLegendUrl:  DS.attr('string'),
  group: DS.belongsTo('group', { async: true })
});
